cli/internal/core/engine.go

```diff
@@ -385,11 +410,76 @@ func (e *Engine) ValidatePersistentDependencies(graph *graph.CompleteGraph) erro
 	return validationError
 }
 
-// GetResolvedTaskDefinition returns a "resolved" TaskDefinition.
-// Today, it just looks up the task from the root Pipeline, but in the future
-// we will compose the TaskDefinition from workspaces using the `extends` key.
-func (e *Engine) GetResolvedTaskDefinition(rootPipeline *fs.Pipeline, taskName string, taskID string) (*fs.TaskDefinition, error) {
-	return rootPipeline.GetTask(taskID, taskName)
+// getTaskDefinitionChain gets a set of TaskDefinitions that apply to the taskID.
+// These definitions should be merged by the consumer.
+// TODO(mehulkar): Can we use getPipelineFromWorkspace() here instead of reading configs again and passing around rootPipeline as an arg?
+func (e *Engine) getTaskDefinitionChain(rootPipeline *fs.Pipeline, pkg *fs.PackageJSON, taskID string, taskName string) ([]fs.BookkeepingTaskDefinition, error) {
+	// Start a list of TaskDefinitions we've found for this TaskID
+	taskDefinitions := []fs.BookkeepingTaskDefinition{}
+
+	// Look for the taskDefinition in the root pipeline.
+	// We'll wait to throw errors until the end
+	if rootTaskDefinition, err := rootPipeline.GetTask(taskID, taskName); err == nil {
+		taskDefinitions = append(taskDefinitions, *rootTaskDefinition)
+	}
+
+	// If the taskID is a root task (e.g. //#build), we don't need to look
+	// for a workspace task, since these can only be defined in the root turbo.json.
+	taskIDPackage, _ := util.GetPackageTaskFromId(taskID)
+	if taskIDPackage != util.RootPkgName && taskIDPackage != ROOT_NODE_NAME {
+		// Look up task definition in turbo.json in the workspace directory
+		workspaceConfigPath := pkg.Dir.Join("turbo.json").RestoreAnchor(e.completeGraph.RepoRoot)
+
+		// If there is an error, we can ignore it, since turbo.json config is not required in the workspace.
+		if workspaceTurboJSON, err := fs.ReadTurboConfig(workspaceConfigPath); err == nil {
+			// TODO(mehulkar): Enable extending from more than one workspace.
+			if len(workspaceTurboJSON.Extends) > 1 {
+				return nil, fmt.Errorf(
+					"You can only extend from the root workspace. \"%s\" extends from %v",
+					pkg.Name,
+					workspaceTurboJSON.Extends,
+				)
+			}
+
+			// TODO(mehulkar):
```

Comment:
Does this mean that defining `pkg#task` in workspace `turbo.json` will work right now? Could this code be replaced with 
```go
if workspaceDefinition, ok := workspaceTurboJSON.Pipeline[taskName]; ok {
  taskDefinitions = append(taskDefinitions, workspaceDefinition)
}
```

Reply:
This is part of the defered work noted above. I have a separate task for it, since it's kind of an edge case.